### PR TITLE
Include all .xml files in targeting pack

### PR DESCRIPTION
**Pull request by wtgodbe**
Resolve the 5.0 part of https://github.com/dotnet/aspnetcore/issues/26073. We were intentionally including the .xml files for project references & Extensions references, but not other package references. 

@Pilchie is this OK for GA tell-mode?

### CODE REVIEW

**Comment by Pilchie** [comment]
Yes, I'm happy with that.

**Comment by dougbu** [comment]
Please retarget to release/5.0-rc2

**Review by dagood — APPROVED** [review]

File: src/Framework/App.Ref/src/Microsoft.AspNetCore.App.Ref.csproj [hunk 1]
```diff
@@ -140,9 +140,13 @@ This package is an internal implementation of the .NET Core SDK and is not meant
...
           Condition="Exists('%(RootDir)%(Directory)%(FileName).xml')" />
       <AspNetCoreReferenceDocXml Include="@(_SelectedExtensionsRefAssemblies->'$(RuntimeExtensionsReferenceDirectory)%(FileName).xml')" />
+      <!-- Grab remaining .xml files from packages -->
+      <AspNetCoreReferenceDocXml 
+          Include="@(AspNetCoreReferenceAssemblyPath->WithMetadataValue('ExternallyResolved', 'true')->'%(RootDir)%(Directory)%(Filename).xml')" 
+          Condition="Exists('%(RootDir)%(Directory)%(Filename).xml')" />
 
       <RefPackContent Include="@(AspNetCoreReferenceAssemblyPath)" PackagePath="$(RefAssemblyPackagePath)" />
-      <RefPackContent Include="@(AspNetCoreReferenceDocXml)" PackagePath="$(RefAssemblyPackagePath)" />
+      <RefPackContent Include="@(AspNetCoreReferenceDocXml->Distinct())" PackagePath="$(RefAssemblyPackagePath)" />
```
Comment (dougbu) [inline src/Framework/App.Ref/src/Microsoft.AspNetCore.App.Ref.csproj:149]: Same comment as in the 3.1 version of this fix

**Comment by wtgodbe** [comment]
> Please retarget to release/5.0-rc2

Done. @Pilchie fine for rc2 tell mode?

**Comment by Pilchie** [comment]
Yes.

**Comment by dougbu** [comment]
https://dev.azure.com/dnceng/_apis/resources/Containers/5139262/Windows_Packages?itemPath=Windows_Packages%2FRelease%2FShipping%2FMicrosoft.AspNetCore.App.Ref.5.0.0-ci.nupkg looks good (ignoring dotnet/runtime#42479)❕ 

image

File: src/Framework/App.Ref/src/Microsoft.AspNetCore.App.Ref.csproj (hunk 1 above)
Comment (dougbu) [reply src/Framework/App.Ref/src/Microsoft.AspNetCore.App.Ref.csproj:149]: Given discussion of this in #26146, is the `->Distinct()` needed❔ If yes, why❔

**Review by dougbu — APPROVED** [review]

**Pull request merged**